components/binary_sensor/rc522.rst

```diff
@@ -0,0 +1,153 @@
+RC522 RFID
+==========
+
+.. seo::
+    :description: Instructions for setting up RC522 RFID tag readers and tags in ESPHome
+    :image: rc522.jpg
+    :keywords: RC522, RFID
+
+.. _rc522-component:
+
+Component/Hub
+-------------
+
+The ``rc522`` component allows you to use RC522 RFID controllers
+(`datasheet <hthttps://www.nxp.com/docs/en/data-sheet/MFRC522.pdff>`__, `Ali Express <https://es.aliexpress.com/item/1260729519.html>`__)
+with ESPHome. This component is a global hub that establishes the connection to the RC522 via :ref:`SPI <spi>` and
+outputs its data. Using the :ref:`RC522 binary sensors <rc522-tag>` you can then
+create individual binary sensors that track if an RFID tag is currently detected by the RC522.
+
+.. figure:: images/rc522-full.jpg
+    :align: center
+    :width: 60.0%
+
+See :ref:`rc522-setting_up_tags` for information on how to setup individual binary sensors for this component.
+
+As the communication with the RC522 is done using SPI you need to have an :ref:`SPI bus <spi>` in your configuration with both
+the **miso_pin** and **mosi_pin** set.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    spi:
+      clk_pin: D0
+      miso_pin: D1
+      mosi_pin: D2
+
+    rc522_spi:
+      cs_pin: D3
+      update_interval: 1s
+
+    binary_sensor:
+      - platform: rc522
+        uid: 74-10-37-94
+        name: "RC522 RFID Tag"
+
+Configuration variables:
+************************
+
+- **cs_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): The pin on the ESP that the chip select line
+  is connected to.
+- **reset_pin** (*Optional*, :ref:`Pin Schema <config-pin_schema>`): The pin connected to the RST line. Some tests
+  shows the RC522 working okay without this.
+- **update_interval** (*Optional*, :ref:`config-time`): The duration of each scan on the RC522. This affects the
+  duration that the individual binary sensors stay active when they're found.
+  If a device is not found within this time window, it will be marked as not present. Defaults to 1s.
+- **on_tag** (*Optional*, :ref:`Automation <automation>`): An automation to perform
+  when a tag is read. See :ref:`rc522-on_tag`.
+- **spi_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the :ref:`SPI Component <spi>` if you want
+  to use multiple SPI buses.
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID for this component.
+
+.. _rc522-on_tag:
+
+``on_tag``
+----------
+
+This automation will be triggered when the RC522 module responds with a tag. Please note that this
+can be called quite often (with an interval of ``update_interval``) as it's triggered repeatedly
+if the tag is re-read many times.
+
+The parameter ``x`` this trigger provides is of type ``std::string`` and is the tag UID in the format
+``74-10-37-94``. The configuration below will for example publish the tag ID on the MQTT topic ``rc522/tag``.
+
+.. code-block:: yaml
+
+    rc522:
```

Comment:
Is this still correct, or should it be `rc522_spi` after the rename of the component?

Reply:
Yes, should be rc522_spi, actually should be something generic like rc522_... (once i2c support docs are done)